File: 프로그래머스/0/181854. 배열의 길이에 따라 다른 연산하기/배열의 길이에 따라 다른 연산하기.py
def solution(arr, n):
    answer = []
    
    if len(arr)%2==1:
        for i,num in enumerate(arr):
            if i%2==0:
                answer.append(num+n)
            else:
                answer.append(num)
    else:
        for i,num in enumerate(arr):
            if i%2==1:
                answer.append(num+n)
            else:
                answer.append(num)
                
    
    return answer

File: 프로그래머스/0/181854. 배열의 길이에 따라 다른 연산하기/배열의 길이에 따라 다른 연산하기.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> arr, int n) {
    vector<int> answer;
    int size=arr.size();
    
    if(size%2==1)
    {
        for(int i=0;i<size;i+=2)
        {
            arr[i]+=n;
        }
    }
    else
    {
        for(int i=1;i<size;i+=2)
        {
            arr[i]+=n;
        }
    }
    answer=arr;
    
    return answer;
}